src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -58,8 +76,11 @@
   private String description;
   private String frameworkName;
   private String slavesUser;
-  private String principal;
-  private String secret;
+  private String credentialsId;
+  @Deprecated
+  private transient String principal;
+  @Deprecated
+  private transient String secret;
```

Comment:
I don't think I want to track these.
* I don't think the setters are being used at the moment
* I'd rather change setters implementations to throw Exceptions to prevent anyone using them, or even delete the methods to force potential client to use the credentials management instead.